snapcraft/tests/fixture_setup.py

```diff
@@ -754,16 +753,20 @@ def setUp(self):
                     'internal', 'repo', 'manifest.txt'))) as manifest_file:
             self.add_packages([line.strip() for line in manifest_file])
 
+    def add_package(self, package):
+        package.temp_dir = self.path
+        self.cache[package.name] = package
```

Comment:
Can you create the FakeaptCachePackage here instead of every call to the function?

Reply:
I wouldn't want the add_package method to receive all the arguments that FakeAptCachePackage. If you want to just add a package by name you can `self.add_packages([test])`. If you need more control than the name, I think it's ok that you instantiate your own.

Maybe that other method should be called `add_packages_by_name`.